File: src/main/java/com/crud_javatokotiln/global/ApiResponse.java
package com.crud_javatokotiln.global;

public record ApiResponse<T>(
        boolean isSuccess,
        T data,
        String message
) {
    public static <T> ApiResponse<T> ok(T data) {
        return new ApiResponse<>(true, data, "Success");
    }

    public static <T> ApiResponse<T> created(T data) {
        return new ApiResponse<>(true, data, "Created");
    }

    public static <T> ApiResponse<T> fail(String message) {
        return new ApiResponse<>(false, null, message);
    }
}

File: src/main/java/com/crud_javatokotiln/global/ApiResponse.kt
package com.crud_javatokotiln.global

import com.fasterxml.jackson.annotation.JsonProperty

data class ApiResponse<T>(
    @get:JsonProperty("isSuccess")
    val isSuccess: Boolean,
    val data: T?,
    val message: String
) {
    companion object {
        @JvmStatic
        fun <T> ok(data: T): ApiResponse<T> = ApiResponse(true, data, "Success")

        @JvmStatic
        fun <T> created(data: T): ApiResponse<T> = ApiResponse(true, data, "Created")

        @JvmStatic
        fun <T> fail(message: String): ApiResponse<T> = ApiResponse(false, null, message)
    }
}
